scylla/src/transport/topology.rs

```diff
@@ -320,8 +326,25 @@ impl MetadataReader {
         res
     }
 
-    fn update_known_peers(&mut self, metadata: &Metadata) {
-        self.known_peers = metadata.peers.iter().map(|peer| peer.address).collect();
+    async fn update_known_peers(&mut self, metadata: &Metadata) {
+        self.known_peers = try_join_all(metadata.peers.iter().map(|peer| async {
```

Comment:
You are using `try_join_all` but all futures it polls always return `Ok`, right? If so, then please use `join_all` instead.